src/test/java/uk/gov/beis/els/apiintegration/RateLimitingIntegrationTest.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.beis.els.apiintegration;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import uk.gov.beis.els.service.Bucket4JRequestService;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Import(value = RateLimitingIntegrationTest.TestConfig.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@ActiveProfiles("bucket4j-integration-test")
+public class RateLimitingIntegrationTest {
+
+  private MockMvc mockMvc;
+
+  @Autowired
+  protected WebApplicationContext context;
+
+  @Autowired
+  protected Bucket4JRequestService bucket4JRequestService;
+
+  @Before
+  public void setUp() throws Exception {
+    mockMvc = MockMvcBuilders
+        .webAppContextSetup(context)
+        .build();
+  }
+
+  @Test
+  public void withSingleIpAddress() throws Exception {
```

Comment:
Coming back to this, adding the stub for now

Reply:
@jamesbarnett91 Added a TODO and created a backog item to come back to this so will merge this now